pkg/kyverno/test/test_command.go

```diff
@@ -573,32 +595,181 @@ func getFullPath(paths []string, policyResourcePath string, isGit bool) []string
 }
 
 func applyPoliciesFromPath(fs billy.Filesystem, policyBytes []byte, valuesFile string, isGit bool, policyResourcePath string, rc *resultCounts) (err error) {
+	versionedvalues := &versionedTest{}
+
+	if err := json.Unmarshal(policyBytes, versionedvalues); err != nil {
+		return sanitizederror.NewWithError("failed to decode yaml", err)
+	}
+
+	// check for old format
+	if versionedvalues.APIVersion == "" {
+		err = nonVersionedPath(fs, policyBytes, valuesFile, isGit, policyResourcePath, rc)
+	} else {
+		err = versionedPath(fs, policyBytes, valuesFile, isGit, policyResourcePath, rc)
+	}
+
+	return err
+
+}
+
+func versionedPath(fs billy.Filesystem, policyBytes []byte, valuesFile string, isGit bool, policyResourcePath string, rc *resultCounts) (err error) {
 	openAPIController, err := openapi.NewOpenAPIController()
+	if err != nil {
+		return sanitizederror.NewWithError("failed to get new openAPIcontroller", err)
+	}
 	engineResponses := make([]*response.EngineResponse, 0)
 	var dClient *client.Client
-	values := &Test{}
+	versionedvalues := &versionedTest{}
 	var variablesString string
 	var pvInfos []policyreport.Info
 	var resultCounts common.ResultCounts
+	var resultsMap map[string]report.PolicyReportResult
+	var testResults []TestResults
+	var policyFullPath []string
+	var resourceFullPath []string
 	store.SetMock(true)
 
-	if err := json.Unmarshal(policyBytes, values); err != nil {
+	if err := json.Unmarshal(policyBytes, versionedvalues); err != nil {
 		return sanitizederror.NewWithError("failed to decode yaml", err)
 	}
 
+	fmt.Printf("\nExecuting %s...", versionedvalues.MetaData.Name)
```

Comment:
Do we need this Printf?

Reply:
We could make a log of that if you'd suggest. It is for the user's understanding.  
```
Executing test-custom-funcs...
applying 1 policy to 2 resources... 
```
This is what it would look like.
So it isn't needed as such, but would be helpful anyway.

Reply:
I see - can you add a comment that this is for user visibility in the cli?

Reply:
Sure!